### PR TITLE
Implement get enterprise connect authorization url

### PR DESCRIPTION
**Pull request by Haziba**
**Notes to reviewer:**
None

### CODE REVIEW

File: cronofy.php [hunk 1]
```diff
@@ -148,6 +148,33 @@ public function getAuthorizationURL($params)
...
+          avoid_linking : Boolean when true means we will avoid linking calendar accounts together under one set of credentials. OPTIONAL
+
+          Response :
+          $url : String. The URL to authorize your enterprise connect access to the Cronofy API
+         */
+
+        $scope_list = join(" ", $params['scope']);
+        $delegated_scope_list = join(" ", $params['delegated_scope']);
+
+        $url = "https://app.cronofy.com/enterprise_connect/oauth/authorize?response_type=code&client_id=" . $this->client_id . "&redirect_uri=" . urlencode($params['redirect_uri']) . "&scope=" . $scope_list . "&delegated_scope=" . $delegated_scope_list;
```
Comment (gshutler) [inline cronofy.php]: Scope lists should be `rawurlencode()` to escape the spaces as `%20`. Should probably prefer `rawurlencode()` to `urlencode()` in all cases as it encodes ` ` (space) as `%20` which has slightly wider support (I've known `+` encoding to have problems, the Cronofy API supports both but I consider `%20` to be better practice).

File: cronofy.php [hunk 2]
```diff
@@ -148,6 +148,33 @@ public function getAuthorizationURL($params)
...
+        if (!empty($params['state'])) {
+            $url.="&state=" . $params['state'];
```
Comment (gshutler) [inline cronofy.php]: Should probably `rawurlencode()` this too in case of spaces, etc.

File: cronofy.php [hunk 3]
```diff
@@ -148,6 +148,33 @@ public function getAuthorizationURL($params)
...
+        }
+        if (!empty($params['avoid_linking'])) {
```
Comment (stephenbinns) [inline cronofy.php]: Enterprise connect doesn't need this as it never links accounts

**Comment by Haziba** [comment]
@stephenbinns @gshutler :mag:

**Comment by stephenbinns** [comment]
The avoid linking comment in the method docs can be removed too

**Comment by stephenbinns** [comment]
:shipit:

**Pull request merged**